driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -405,9 +405,52 @@ private void incrementAuthErrorMetric() {
         }
     }
 
-    private UnsupportedProtocolVersionException unsupportedProtocolVersionException(ProtocolVersion triedVersion, ProtocolVersion serverProtocolVersion) {
-        logger.debug("Got unsupported protocol version error from {} for version {} server supports version {}", address, triedVersion, serverProtocolVersion);
-        return new UnsupportedProtocolVersionException(address, triedVersion, serverProtocolVersion);
+    private static final Pattern SUPPORTED_VERSIONS_PATTERN = Pattern.compile("supported versions are \\(([^)]+?)\\)");
```

Comment:
I don't understand, the client starts negotiation with its highest version. If it started with vN, it doesn't support N+1 nor N+2.